[Django + Flask] Getting Started Frameworks section for Upsun (#3816)
platformsh/platformsh-docs: sites/friday/src/get-started/stacks/django.md

    It includes common reference materials useful for deploying Django, but also external community and blog resources that cover more advanced topics relevant for the framework.
---

{{< note theme="info" >}}

Before you proceed, be sure to checkout the [{{% vendor/name %}} demo app](https://console.upsun.com/projects/create-project) and the main [Getting started guide](/get-started/here/_index.md). These two resources provide all of the core concepts and common commands you'll need to know before using the materials below.

{{< /note >}}

Now that you have completed the [Getting started guide](/get-started/here/_index.md), there are **six remaining changes**
required in order to have a successful deployment of Django on Upsun. Alternatively, you can read the blog post
[_Up(sun) and running with Django_](https://upsun.com/blog/setting-up-django-on-upsun/) which walks through the steps
required to run Django on Upsun.

## `.environment`

> Django requires a few environment variables that have not been set. Open the `.environment` file and add the following:

Reviewer: ```suggestion
Your Django app, depending on your `settings.py`, may require environment variables that have not been set. Open the `.environment` file and add them:
```

Reviewer: Not all Django apps will allow for this override AFAIK - pulling them in has to be done explicitly in the `settings.py` file.

```
to
```yaml
start: "gunicorn -b unix:$SOCKET config.wsgi"
```
If your Django instance requires a different Web server, we also support [several other options](/languages/python/server.md).

### Static assets
In order to access Django's static assets we need to add a second location to the `web:locations` block. In the
`.upsun/config.yaml` file, locate the `web:locations` section. Expand what is currently there to include a location for
`/static`:

```yaml
            locations:
              "/":
                "passthru": true
              "/static":
                "allow": true
                "expires": "1h"
                "root": "static"
```

### Installation during the Build Hook
Next we need to instruct Upsun to install our python and node dependencies. Scroll down a bit farther and find the
`hooks:build` section. Change the following lines from:
```yaml
    build: |
      set -eux

      # Add build steps here.
      # echo 'I am a build step'
```
to:
```yaml
    build: |
      set -eux
      pip install --upgrade pip
      pip install -r requirements.txt
      npm install
      npm run build
```

> ### Deploy Hook

Reviewer: ```suggestion
If your project uses a different package manager, we also support [several other options](/languages/python/dependencies.md).

### Deploy Hook
```

Just below that, find the `deploy` section and change the following lines from:
```yaml
      deploy: |
        set -eux
        # echo 'Put your deploy command here'
```
to:
```yaml

>             deploy: |

Reviewer 2: indentation changed?

Reviewer 2: maybe you need to display all top level keys above, like; 
```
applications:
    app:
        hooks:
            deploy: |
```

Reviewer 2: and same for all other YAML snippets?

Author: that's the challenging part with yaml. do our code samples include the correct indentation according to where the keys we're changing are located so if someone copies and pastes they get the right indentation (assuming they're using 2 spaces per indent)? Or do we include the entire path (ie `applications:<app-name>:hooks`) and _assume_ the user knows that they cant copy and paste the whole block?

I don't have a good answer.

              set -eux
              python manage.py collectstatic --noinput
              python manage.py migrate
```

### Mounts
Since Django requires a writable location, we'll need to attach writable mount. Locate the `mounts` section (currently
commented). Change the following lines from:

```yaml
    # mounts:
    #   "/.cache": # Represents the path in the app.
    #     source: "local" # "local" sources are unique to the app, while "service" sources can be shared among apps.
    #     source_path: "cache" # The subdirectory within the mounted disk (the source) where the mount should point.
```
to

```yaml

>         mounts:

Reviewer 2: indentation is different...
